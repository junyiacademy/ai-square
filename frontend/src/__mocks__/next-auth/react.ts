export const useSession = jest.fn(() => ({
  data: null,
  status: 'unauthenticated'
}));

export const signIn = jest.fn();
export const signOut = jest.fn();
export const getCsrfToken = jest.fn();
export const getSession = jest.fn();
export const getProviders = jest.fn();

export const SessionProvider = ({ children }: { children: React.ReactNode }) => children;