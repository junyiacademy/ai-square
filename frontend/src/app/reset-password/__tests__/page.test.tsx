/**
 * Tests for page
 */

import React from 'react';
import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import Page from '../page';

// Mock dependencies
jest.mock('next/navigation', () => ({
  useRouter: () => ({
    push: jest.fn(),
    back: jest.fn(),
    refresh: jest.fn(),
  }),
  usePathname: () => '/',
  useSearchParams: () => new URLSearchParams(),
}));

describe('page', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render without crashing', () => {
    const { container } = render(<Page />);
    expect(container).toBeInTheDocument();
  });

  it('should have proper structure', () => {
    render(<Page />);
    
    // Check for the page heading
    expect(screen.getByRole('heading', { name: /resetPassword\.title/i })).toBeInTheDocument();
    
    // Check for the error message
    expect(screen.getByText(/resetPassword\.noToken/i)).toBeInTheDocument();
    
    // Check for the link to request new link
    expect(screen.getByRole('link', { name: /resetPassword\.requestNewLink/i })).toBeInTheDocument();
  });

  it('should handle user interactions', async () => {
    render(<Page />);
    
    // Look for interactive elements
    const buttons = screen.queryAllByRole('button');
    const links = screen.queryAllByRole('link');
    const inputs = screen.queryAllByRole('textbox');
    
    // Test at least one interaction if available
    if (buttons.length > 0) {
      fireEvent.click(buttons[0]);
      // Add assertion based on expected behavior
    }
    
    if (inputs.length > 0) {
      fireEvent.change(inputs[0], { target: { value: 'test' } });
      expect(inputs[0]).toHaveValue('test');
    }
  });

  it('should be accessible', () => {
    const { container } = render(<Page />);
    
    // Basic accessibility checks
    const headings = container.querySelectorAll('h1, h2, h3, h4, h5, h6');
    const images = container.querySelectorAll('img');
    
    images.forEach(img => {
      expect(img).toHaveAttribute('alt');
    });
  });

  it('should match snapshot', () => {
    const { container } = render(<Page />);
    expect(container.firstChild).toMatchSnapshot();
  });
});