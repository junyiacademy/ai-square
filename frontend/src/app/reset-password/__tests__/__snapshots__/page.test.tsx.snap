// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`page should match snapshot 1`] = `
<div
  class="min-h-screen flex items-center justify-center bg-gradient-to-br from-blue-50 to-indigo-100 dark:from-slate-900 dark:to-indigo-900 py-12 px-4 sm:px-6 lg:px-8"
>
  <div
    class="max-w-md w-full space-y-8"
  >
    <div>
      <h2
        class="mt-6 text-center text-3xl font-extrabold text-gray-900 dark:text-white"
      >
        resetPassword.title
      </h2>
    </div>
    <div
      class="rounded-md bg-red-50 dark:bg-red-900/20 p-4"
    >
      <p
        class="text-sm text-red-800 dark:text-red-400"
      >
        resetPassword.noToken
      </p>
      <a
        class="mt-2 inline-block text-sm text-indigo-600 hover:text-indigo-500"
        href="/forgot-password"
      >
        resetPassword.requestNewLink
      </a>
    </div>
  </div>
</div>
`;
